api/users.js

```diff
@@ -1,4 +1,115 @@
-const usersRouter = require("express").Router();
-//insert query requests here//
+const express = require("express");
+const usersRouter = express.Router();
+const { requireUser, checkOwner } = require("./utils.js");
+const jwt = require("jsonwebtoken");
+require("dotenv").config();
+const { JWT_SECRET } = process.env;
+const { createUser, getUser, getUserByUsername, updateUser } = require("../db");
+
+usersRouter.use((req, res, next) => {
+  console.log("A request is being made to /users");
+
+  next();
+});
+
+// POST /users/register
+// Create a new user. Require username and password, and
+// hash password before saving user to DB.
+// Require all passwords to be at least 8 characters long.
+usersRouter.post("/register", async (req, res, next) => {
+  try {
+    const { username, password } = req.body;
```

Comment:
In order for line 37 to work, I think this needs to be changed to:
const { username, password, full_name, email } = req.body;

Reply:
Made update